pkg/cluster/cluster.go

```diff
@@ -817,3 +822,57 @@ func (c *Cluster) shouldDeleteSecret(secret *v1.Secret) (delete bool, userName s
 	secretUser := string(secret.Data["username"])
 	return (secretUser != c.OpConfig.ReplicationUsername && secretUser != c.OpConfig.SuperUsername), secretUser
 }
+
+type simpleActionWithResult func() error
+
+func (c *Cluster) deletePatroniClusterObjects() error {
+	// TODO: figure out how to remove leftover patroni objects in other cases
+	if !c.patroniUsesKubernetes() {
+		c.logger.Infof("not cleaning up Etcd Patroni objects on cluster delete")
+	}
+	c.logger.Debugf("removing leftover Patroni objects (endpoints or configmaps)")
+	for _, deleter := range []simpleActionWithResult{c.deletePatroniClusterEndpoints, c.deletePatroniClusterConfigMaps} {
+		if err := deleter(); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// TODO: figure out the interface shenanigans to avoid code duplication below.
+func (c *Cluster) deletePatroniClusterEndpoints() error {
+	for _, suffix := range patroniObjectSuffixes {
+		configEndpointName := fmt.Sprintf("%s-%s", c.Name, suffix)
+
+		if ep, err := c.KubeClient.Endpoints(c.Namespace).Get(configEndpointName, metav1.GetOptions{}); err == nil {
```

Comment:
Why not pass `c.KubeClient.Endpoints(c.Namespace)` as an argument here? In this way we can avoid this code duplication.

Reply:
I don't think it is that easy. Endpoint and ConfigMap implement 2 different interfaces and Get functions actually return different types (*v1.Endpoint and *v1.ConfigMap and respectively).